File: 백준/code/15651.py
def sequence(k, tmp):
    if k == M:
        print(*tmp); return        
    for i in range(N):
        sequence(k+1, tmp+arr[i])

N, M = map(int, input().split())
arr = list(map(str, range(1, N + 1)))

sequence(0, '')

File: 백준/code/15651.js
const [N, M] = require('fs').readFileSync('/dev/stdin').toString().trim().split(' ').map(Number)

solution(N, M)

function solution(N, M) {
  const permuArr = deplePermutation(N, M)
  console.log(permuArr.map(n => n.join(' ')).join('\n'))
}

function deplePermutation(N, M) {
  const result = []
  const tmp = []

  permu()

  function permu() {
    if (tmp.length === M) {
      result.push([...tmp])
      return
    }

    for (let i = 1; i <= N; i++) {
      tmp.push(i)
      permu()
      tmp.pop()
    }
  }
  return result
}